jmix-ui/ui/src/main/java/io/jmix/ui/model/impl/InstanceLoaderImpl.java

```diff
@@ -71,6 +81,34 @@ public void setDataContext(@Nullable DataContext dataContext) {
         this.dataContext = dataContext;
     }
 
+    @Override
+    public void setMonitoringInfoProvider(Function<DataLoader, DataLoaderMonitoringInfo> monitoringInfoProvider) {
+        Preconditions.checkNotNullArgument(monitoringInfoProvider);
+        this.monitoringInfoProvider = monitoringInfoProvider;
+    }
+
+    @Override
+    public Function<DataLoader, DataLoaderMonitoringInfo> getMonitoringInfoProvider() {
+        return monitoringInfoProvider;
+    }
+
+    protected E loadWithTimer(Callable<E> loader) {
+        E entity;
+
+        Timer.Sample sample = UiMonitoring.startTimerSample(meterRegistry);
+        try {
+            entity = loader.call();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            DataLoaderMonitoringInfo info = monitoringInfoProvider.apply(this);
```

Comment:
In case of exception there can be false "fast" loading metric records which corrupts statistics.